src/multi_vector_simulator/F1_plotting.py

```diff
@@ -855,52 +856,70 @@ def create_plotly_flow_fig(
+                    y=df_plots_data[asset],
+                    mode="lines",
+                    line=dict(color=get_color(i, color_list), width=2.5),
+                    name=asset,
+                )
+            )
 
-    assets_list = list(df_plots_data.columns)
-    assets_list.remove("timestamp")
+        fig.update_layout(
+            xaxis_title=x_legend,
+            yaxis_title=y_legend,
+            font_family="sans-serif",
+            template="simple_white",
+            xaxis=styling_dict,
+            yaxis=styling_dict,
+            title={
+                "text": plot_title,
+                "y": 0.90,
+                "x": 0.5,
+                "font_size": 23,
+                "xanchor": "center",
+                "yanchor": "top",
+            },
+            legend=dict(y=0.5, traceorder="normal", font=dict(color="black"),),
+        )
 
-    for i, asset in enumerate(assets_list):
-        fig.add_trace(
-            go.Scatter(
-                x=df_plots_data["timestamp"],
-                y=df_plots_data[asset],
-                mode="lines",
-                line=dict(color=get_color(i, color_list), width=2.5),
-                name=asset,
+        if file_path is not None:
+            # Function call to save the Plotly plot to the disk
+            save_plots_to_disk(
+                fig_obj=fig,
+                file_path=file_path,
+                file_name=filename,
+                width=1200,
+                height=600,
+                scale=5,
             )
-        )
+        return fig
 
-    fig.update_layout(
-        xaxis_title=x_legend,
-        yaxis_title=y_legend,
-        font_family="sans-serif",
-        template="simple_white",
-        xaxis=styling_dict,
-        yaxis=styling_dict,
-        title={
-            "text": plot_title,
-            "y": 0.90,
-            "x": 0.5,
-            "font_size": 23,
-            "xanchor": "center",
-            "yanchor": "top",
-        },
-        legend=dict(y=0.5, traceorder="normal", font=dict(color="black"),),
-    )
+    assets_list = list(df_plots_data.columns)
+    assets_list.remove("timestamp")
 
-    if file_path is not None:
-        # Function call to save the Plotly plot to the disk
-        save_plots_to_disk(
-            fig_obj=fig,
-            file_path=file_path,
-            file_name=file_name,
-            width=1200,
-            height=600,
-            scale=5,
+    if any(SOC in item for item in assets_list):
+        # plot SOC separately
+        adapted_title = plot_title.replace("power", "storage SOC")
+        create_plot(
+            flow_list=[s for s in assets_list if SOC in s],
+            filename=f"SOC_{file_name}",
+            y_legend="SOC in %",
+            plot_title=adapted_title,
         )
+        # remove SOC as it is provided in % and does not fit to this graph
+        assets_list = [s for s in assets_list if SOC not in s]
+
+    fig = create_plot(flow_list=assets_list)
```

Comment:
Do you intentionally overwrite `fig = go.Figure()`?, in any case, the SOC figure is stored to png with `-png`
![SOC_Electricity_bus_power](https://user-images.githubusercontent.com/44204527/99689889-ce002680-2a87-11eb-8097-61f3c24aa2df.png)
But it is not stored to the autoreport (see pdf above).

Reply:
> Hi Sabine! The SOC plot is not working out yet: It is not stored into the autoreport. I dont really know much about plotly, so you need to ask @Bachibouzouk and @mahendrark about this.

I see.. I don't know much about it either, which is why it didn't work out automatically. Will fix this.

> 
> Is there no test you can adapt for your changes in `A1.add_storage_components()`? Can you create a test that only tests what you want it to test (storage subasset label) easily?
> 
> Is there no test you can adapt for your changes in `E1.get_timeseries_per_bus()`? Can you create a test that checks only your changes easily?

There are no tests for these functions, yet. I'll add tests for my changes and if it's fast also more..